core/templates/services/improvements.service.ts

```diff
@@ -29,6 +29,9 @@ export class ImprovementsService {
   INTERACTION_IDS_REQUIRED_TO_BE_RESOLVED = ['TextInput'];
 
   isStateForcedToResolveOutstandingUnaddressedAnswers(state: State): boolean {
+    if (state.interaction.id === null) {
```

Comment:
It seems odd to check for this condition before you check for `!!state` (which the return statement below does), right?

Reply:
Hmm, why are we exactly using `!!state` wouldn't this be equivalent to `state` ?
Does below function seems correct ?
```typescript
  isStateForcedToResolveOutstandingUnaddressedAnswers(state: State): boolean {
    if (state && state.interaction.id === null) {
      return false;
    }
    return this.INTERACTION_IDS_REQUIRED_TO_BE_RESOLVED.indexOf(
      state.interaction.id) !== -1;
  }
```

Reply:
`!!state` is a typical way in JavaScript to cast to a boolean value. So you'll probably want to keep that.

I don't think the logic is quite correct. You'll basically want to return false if either the state is null or the interaction ID is null.

Reply:
Yeah, both are equivalent. See https://stackoverflow.com/questions/29312123/how-does-the-double-exclamation-work-in-javascript

I agree that Boolean(...) is probably clearer, though `!!` and `!` are fairly idiomatic/standard in industry JavaScript.